### PR TITLE
Continue In ProcessPreGenesisDeposits

### PR DESCRIPTION
**Pull request by rauljordan**
Part of #7410, this PR returns an error in `ProcessPreGenesisDeposits` if a pubkey for a deposit is not found in the beacon state. This is a good best practice as outlined by trail of bits. However, the conditional will never trigger because of how the function works:

1. Process all new deposits into the beacon state
2. Then, for each deposit that was processed, fetch the validator corresponding to the public key in the deposit

Since step (1) added new validators into the state before (2) runs, there will _always_ be a corresponding validator for a public key for each processed deposits. As such, it is impossible to write a test for this. However, we should still handle the error for consistency.

### CODE REVIEW

**Review by terencechain — APPROVED** [review]

**Review by nisdas — CHANGES_REQUESTED** [review]
This is not correct, we need to continue processing in the even if the validator does not exist. There is always the possibility of processing invalid deposits with bad pubkeys/signatures.

File: beacon-chain/core/blocks/deposit_test.go [hunk 1]
```diff
@@ -272,3 +272,71 @@ func TestProcessDeposit_SkipsInvalidDeposit(t *testing.T) {
...
+		require.Equal(t, uint64(0), val.ActivationEligibilityEpoch)
+	}
+	if newState.Eth1DepositIndex() != 100 {
+		t.Errorf(
+			"Expected Eth1DepositIndex to be increased by 99 after processing an invalid deposit, received change: %v",
+			newState.Eth1DepositIndex(),
+		)
+	}
+	if len(newState.Validators()) != 100 {
+		t.Errorf("Expected validator list to have length 1, received: %v", len(newState.Validators()))
```
Comment (shayzluf) [inline beacon-chain/core/blocks/deposit_test.go:334]: seems like you are expecting 100 not 1 as the comment mentions

Comment (rauljordan) [inline beacon-chain/core/blocks/deposit_test.go:334]:
```suggestion
		t.Errorf("Expected validator list to have length 100, received: %v", len(newState.Validators()))
```

File: beacon-chain/core/blocks/deposit_test.go [hunk 2]
```diff
@@ -272,3 +272,71 @@ func TestProcessDeposit_SkipsInvalidDeposit(t *testing.T) {
...
+	}
+	if len(newState.Balances()) != 100 {
+		t.Errorf("Expected validator balances list to have length 1, received: %v", len(newState.Balances()))
```
Comment (rauljordan) [inline beacon-chain/core/blocks/deposit_test.go:337]:
```suggestion
		t.Errorf("Expected validator balances list to have length 100, received: %v", len(newState.Balances()))
```

**Review by shayzluf — APPROVED** [review]
Lgtm

**Pull request merged**